src/main/java/com/github/mtsochacki/adventofcode/Day18.java

```diff
@@ -0,0 +1,183 @@
+package com.github.mtsochacki.adventofcode;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Day18 implements Day {
+    @Data
+    @AllArgsConstructor
+    private static class RegularNumber {
+        private int level;
+        private int value;
+
+        public RegularNumber(RegularNumber regularNumber) {
+            this.level = regularNumber.level;
+            this.value = regularNumber.value;
+        }
+
+        public void incrementLevel() {
+            this.level++;
+        }
+
+        public void decrementLevel() {
+            this.level--;
+        }
+
+        public void increaseValue(int value) {
+            this.value += value;
+        }
+    }
+
+    @Override
+    public String part1(List<String> input) {
+        Integer slugMagnitude = input.stream()
+                .map(this::transformIntoSnailfishNumber)
+                .reduce(this::reduceSnailfishNumber)
+                .map(this::calculateMagnitude)
+                .orElseThrow(() -> new RuntimeException("Error processing SnailFish"));
+        return String.valueOf(slugMagnitude);
+    }
+
+    @Override
+    public String part2(List<String> input) {
+        List<List<RegularNumber>> snailfishNumbers = input.stream()
+                .map(this::transformIntoSnailfishNumber)
+                .toList();
+        int max = findHighestMagnitude(snailfishNumbers);
+        return String.valueOf(max);
+    }
+
+    private int findHighestMagnitude(List<List<RegularNumber>> snailfishNumbers) {
+        int max = 0;
+        for (int i = 0; i < snailfishNumbers.size(); i++) {
+            for (int j = 0; j < snailfishNumbers.size(); j++) {
```

Comment:
I think you're considering each pair twice here: think about e.g `i = 1, j = 3` and `i = 3, j = 1`. I'd suggest either starting `j` from `i + 1` or not doing the `secondResult` calculation at all.

Reply:
Yeah, that was a bit silly :D
I am removing `secondResult` altogether, I think this way it is going to be easier to comprehend what is going on

Reply:
I agree that's nicer in this case